consensus/polybft/polybft.go

```diff
@@ -141,8 +141,12 @@ func GenesisPostHookFactory(config *chain.Chain, engineName string) func(txn *st
 			return err
 		}
 
-		if err = mintRewardTokensToWalletAddress(&polyBFTConfig, transition); err != nil {
-			return err
+		if !isNativeRewardToken(polyBFTConfig) {
```

Comment:
Nitpicking: Why don't just introduce this check `if !isNativeRewardToken(polyBFTConfig)` at the beginning of  `mintRewardTokensToWalletAddress`. Then `mintRewardTokensToWalletAddress` is safe? Maybe I am worng, just asking.

Reply:
I was first thinking putting it there, but then I thought why it should even call the function if it is native reward token and immediately jump out of it. But since you have asked the same, I'd consider putting condition within the mint function itself (although it is expected to be called only on this place).

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1594/commits/7557c1d623769385a0ebe48f3acbc6e4461fbbdb